src/main/java/core/gui/model/PlayerOverviewTableModel.kt

```diff
@@ -0,0 +1,223 @@
+     */
+    private fun getPreviousPlayerDevelopmentStage(currentDevelopmentStage: Player): Player? {
+        val id = currentDevelopmentStage.playerId
+
+        val selectedPlayerDevelopmentStage = SpielerTrainingsVergleichsPanel.getSelectedPlayerDevelopmentStage()
+        var i = 0
+        while ((selectedPlayerDevelopmentStage != null) && (i < selectedPlayerDevelopmentStage.size)) {
+            val selectedDevelopmentStage = selectedPlayerDevelopmentStage[i]
+
+            if (selectedDevelopmentStage.playerId == id) {
+                return selectedDevelopmentStage
+            }
+            i++
+        }
+
+        if (SpielerTrainingsVergleichsPanel.isDevelopmentStageSelected()) {
+            val hrf = SpielerTrainingsVergleichsPanel.getSelectedHrfId()
+            return getFirstPlayerDevelopmentStageAfterSelected(currentDevelopmentStage, hrf)
+        }
+        return null
+    }
+
+    /**
+     * Returns the [Player] from the first HRF in which he appears.
+     */
+    private fun getFirstPlayerDevelopmentStageAfterSelected(vorlage: Player, hrfId: Int?): Player {
+        var after: HODateTime? = null
+        if (hrfId != null) {
+            val hrf = DBManager.instance().loadHRF(hrfId)
+            if (hrf != null) {
+                after = hrf.datum
+            }
+        }
+        return DBManager.instance().loadPlayerFirstHRF(vorlage.playerId, after)
+    }
+
+    /**
+     * create a data[][] from player-Vector
+     */
+    override fun initData() {
+        val tmpDisplayedColumns = getDisplayedColumns()
+        m_clData = Array(players!!.size) { arrayOfNulls(tmpDisplayedColumns.size) }
+
+        for (i in players!!.indices) {
+            val currentPlayer = players!![i]
+            val comparisonPlayer = getPreviousPlayerDevelopmentStage(currentPlayer)
+            for (j in tmpDisplayedColumns.indices) {
+                if (tmpDisplayedColumns[j] is PlayerColumn) {
+                    m_clData!!.get(i)[j] = (tmpDisplayedColumns[j] as PlayerColumn).getTableEntry(currentPlayer, comparisonPlayer)
+                } else if (tmpDisplayedColumns[j] is BooleanColumn) {
+                    m_clData!!.get(i)[j] = (tmpDisplayedColumns[j] as BooleanColumn).getValue(currentPlayer)
+                }
+            }
+        }
+
+        fireTableDataChanged()
+    }
+
+    /**
+     * Initializes the lineup only
+     */
+    fun reInitData() {
+        val tmpDisplayedColumns = getDisplayedColumns()
+        for (i in players!!.indices) {
+            val currentPlayer = players!![i]
+            for (j in tmpDisplayedColumns.indices) {
+                if (tmpDisplayedColumns[j].id == UserColumnFactory.NAME || tmpDisplayedColumns[j].id == UserColumnFactory.LINEUP || tmpDisplayedColumns[j].id == UserColumnFactory.BEST_POSITION || tmpDisplayedColumns[j].id == UserColumnFactory.SCHUM_RANK_BENCHMARK || tmpDisplayedColumns[j].id == UserColumnFactory.GROUP) {
```

Comment:
Would you mind breaking up into multiple lines for readability?

Reply:
done